test/Swashbuckle.AspNetCore.Cli.Test/ToolTests.cs

```diff
@@ -35,6 +35,50 @@ public void Can_Generate_Swagger_Json()
             }
         }
 
+
+        [Fact]
+        public void CustomDocumentSerializer_Writes_Custom_V2_Document()
+        {
+            var dir = Directory.CreateDirectory(Path.Join(Path.GetTempPath(), Path.GetRandomFileName()));
+            try
+            {
+                var args = new string[] { "tofile", "--output", $"{dir}/swagger.json", "--serializeasv2", Path.Combine(Directory.GetCurrentDirectory(), "CustomDocumentSerializer.dll"), "v1" };
+                Assert.Equal(0, Program.Main(args));
+
+                using var document = JsonDocument.Parse(File.ReadAllText(Path.Combine(dir.FullName, "swagger.json")));
+
+                // verify that the custom serializer wrote the swagger info
+                var swaggerInfo = document.RootElement.GetProperty("swagger").GetString();
+                Assert.Equal("DocumentSerializerTest2.0", swaggerInfo);
+            }
+            finally
+            {
+                dir.Delete(true);
```

Comment:
Might be useful to "borrow" this utility class I wrote recently and add it to the test helpers to use in these tests: [TemporaryDirectory](https://github.com/martincostello/dotnet-bumper/blob/main/src/DotNetBumper.Core/TemporaryDirectory.cs)

Reply:
Added a version of it to Swashbuckle.AspNetCore.TestSupport